### ISSUE TITLE
MapLane didReceive not being called

### ISSUE THREAD

**Issue by SirCipher** [swimos/swim#31]
`MapDownlinkSpec#testDrop` is not calling the `didReceive` phase or the values are not being set. A latch has been used to await the downlink's synchronisation but the error still occurs after a number of runs.

```
expected [0] but found [2]
java.lang.AssertionError: expected [0] but found [2]
	at testng@6.14.3/org.testng.Assert.fail(Assert.java:96)
	at testng@6.14.3/org.testng.Assert.failNotEquals(Assert.java:776)
	at testng@6.14.3/org.testng.Assert.assertEqualsImpl(Assert.java:137)
	at testng@6.14.3/org.testng.Assert.assertEquals(Assert.java:118)
	at testng@6.14.3/org.testng.Assert.assertEquals(Assert.java:547)
	at testng@6.14.3/org.testng.Assert.assertEquals(Assert.java:557)
	at swim.server/swim.server.MapDownlinkSpec.testDrop(MapDownlinkSpec.java:437)
	at jdk.internal.reflect.GeneratedMethodAccessor3.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at testng@6.14.3/org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at testng@6.14.3/org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at testng@6.14.3/org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at testng@6.14.3/org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at testng@6.14.3/org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at testng@6.14.3/org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at testng@6.14.3/org.testng.TestRunner.privateRun(TestRunner.java:648)
	at testng@6.14.3/org.testng.TestRunner.run(TestRunner.java:505)
	at testng@6.14.3/org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at testng@6.14.3/org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at testng@6.14.3/org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at testng@6.14.3/org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at testng@6.14.3/org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at testng@6.14.3/org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at testng@6.14.3/org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at testng@6.14.3/org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at testng@6.14.3/org.testng.TestNG.runSuites(TestNG.java:1049)
	at testng@6.14.3/org.testng.TestNG.run(TestNG.java:1017)
	at org.gradle.api.internal.tasks.testing.testng.TestNGTestClassProcessor.runTests(TestNGTestClassProcessor.java:141)
	at org.gradle.api.internal.tasks.testing.testng.TestNGTestClassProcessor.stop(TestNGTestClassProcessor.java:90)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:132)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:175)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:157)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.base/java.lang.Thread.run(Thread.java:835)
```

**Comment by DobromirM** [comment]
Has this been fixed by #32?

**Comment by ajay-gov** [comment]
Closing, since this has been fixed.
